### ISSUE TITLE
SMTP doesn't accept RCPT with local part > 64 octets.

### ISSUE THREAD

**Issue by msapiro** [aio-libs/aiosmtpd#257]
smtp.py in the code in _getaddr() at lines 1097 - 1118 contains at lines 1115,1116
```
        if len(localpart) > 64:  # RFC 5321 § 4.5.3.1.1
            return None, None
```
While it is true that RFC 5321 § 4.5.3.1.1 specifies 64 octets as the maximum length of a local part, § 4.5.3.1
says in part `To the  maximum extent possible, implementation techniques that impose no  limits on the length of these objects should be used.`

Also, the stdlib smtpd.py _getaddr() contains no such restriction.

Thus, I think SMTP should be more forgiving since I see no reason to enforce this limit.

This restriction results in issues for GNU Mailman which uses the LMTP class to receive messages and receives confirmation messages addressed to `LISTNAME-confirm+TOKEN@example .com. The length of the token is 40 bytes which means if the length of the list name is > 15 bytes, the local part will exceed 64 bytes and be rejected.

**Comment by pepoluan** [comment]
Thank you! It's inputs like this that are greatly appreciated, because we are juggling between strict SMTP compliance and real-world usage.

I'll push 1.4.1 later today.

**Comment by pepoluan** [comment]
@msapiro I've merged the patch, will release 1.4.1 later today.

**Comment by msapiro** [comment]
@pepoluan That's great. I really appreciate your prompt action on this.
